client/extensions/woocommerce/components/product-search/variations.js

```diff
@@ -0,0 +1,102 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { filter, forEach, kebabCase, keys, sortBy } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormLegend from 'components/forms/form-legend';
+import FormSelect from 'components/forms/form-select';
+import Notice from 'components/notice';
+
+// Use a constant for the default attribute state.
+const DEFAULT_ATTR = 'any';
+
+class ProductVariations extends Component {
+	static propTypes = {
+		onChange: PropTypes.func,
+		product: PropTypes.oneOfType( [ PropTypes.bool, PropTypes.object ] ),
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = {};
+		const attributes = filter( props.product.attributes, { variation: true } );
+		forEach( attributes, attr => {
+			this.state[ attr.name ] = DEFAULT_ATTR;
+		} );
+	}
+
+	onChange = name => event => {
+		this.state[ name ] = event.target.value;
+		this.props.onChange( this.state, this.resetAttrs );
+	};
+
+	resetAttrs = () => {
+		this.setState( prevState => {
+			const newState = {};
+			keys( prevState ).map( name => {
+				newState[ name ] = DEFAULT_ATTR;
+			} );
+			return newState;
+		} );
+	};
+
+	renderAttribute = attribute => {
+		const { translate } = this.props;
+		return (
+			<div className="product-search__variation-field" key={ attribute.id }>
+				<FormLabel htmlFor={ `select-${ kebabCase( attribute.name ) }` }>
+					{ attribute.name }
+				</FormLabel>
+				<FormSelect
+					id={ `select-${ kebabCase( attribute.name ) }` }
+					onChange={ this.onChange( attribute.name ) }
+					value={ this.state[ attribute.name ] }
+				>
+					<option key={ DEFAULT_ATTR } value={ DEFAULT_ATTR }>
+						{ translate( 'Select one' ) }
+					</option>
+					{ attribute.options.map( ( opt, i ) => <option key={ i }>{ opt }</option> ) }
```

Comment:
I think that `i` as a key isn't specific enough when you have more than one attribute. I'm getting a console warning on the same key `0` when I have two attributes on a variable product.

Reply:
Strange, I'm not. But I'll update it regardless, seems like a good idea.